### PR TITLE
Updates phenotype hive import -e100

### PR DESCRIPTION
**Pull request by ima23**
- gwas import: update variation sets at import + update display flag at import + evidence attribs 
- checks: fix & update regex
- check if seq_regions exist, otherwise insert them
- refactor hive structure and flow
- do not remove phenotypes when cleaning the existing phenotype features
- add data types to sources
- bug fixes

### CODE REVIEW

File: modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/BasePhenotypeAnnotation.pm [hunk 1]
```diff
@@ -1396,4 +1535,129 @@ sub _iri2acc{
...
+
+  Arg [1]    : Bio::EnsEMBL::DBSQL::DBConnection $dbc
+               The new variation database connection
+  Arg [2]    : string $st
+               The SQL statement to be run.
+  Example    : $obj->_count_results($dbc, $st)
+  Description: Takes SQL statements to count the rows in a table or count rows grouped
+               by an attribute in the table. It returns either the total number of rows in the
+               table or a hash of attribute => row count depending on input.
+  Returntype : interger or hashref
```
Comment (helensch) [inline modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/BasePhenotypeAnnotation.pm]: Typo on interger

**Comment by ima23** [comment]
Thanks for the PR review @helensch.
I have updated the PR with fixes and improvements to the comments, shorter lines,  bracketing open die statement, and constants all caps. I have also created jira tickets for e.g. for update meta reuse.

> Overall the PR is fine. Most comments are documentation and style comments e.g. shorter lines.
> Main suggestion is that constants are in capitals.
> 
> Minor suggestions:
> 
> * strftime used with same format, could this be a sub
> * updating meta could be  sub used for several pipelines
> * bracketing in open and die, could be closer to function name open( instead of open (, as for other subs

File: modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/PhenotypeAnnotation_conf.pm [hunk 2]
```diff
@@ -372,6 +429,123 @@ sub pipeline_analyses {
...
+                @common_params,
+            },
+            -input_ids      => [],
+            -hive_capacity  => 1,
+            -rc_name    => 'default',
+            -flow_into  => {
+                '2' => [ 'import_omia' ],
+                '3' => [ 'import_animalqtldb' ],
+            },
+            -failed_job_tolerance => 5, # tries 5 times to run a job
```
[the remark this answers is not in the document]
Comment (ima23) [reply modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/PhenotypeAnnotation_conf.pm]: @helensch : I probably got the line from somewhere, can't find it where. Yes, `max_retry_count` is the correct retry count option. Will update this in the next commit.

File: modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/ImportHuman.pm [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    my $self = shift;
+
+    my $pipeline_dir = $self->required_param('pipeline_dir');
+    my $run_type = $self->required_param('run_type');
+
+    open (my $pipelogFH, ">", $pipeline_dir."/".'log_import_debug_pipe_human') || die ("Failed to open file: $!\n");
+    $self->pipelogFH($pipelogFH);
+
+    unless ($run_type eq NONE) {
+      my %import_species = &species;
```
[the remark this answers is not in the document]
Comment (ima23) [reply modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/ImportHuman.pm]: It is a constant, no need for &. Constant naming fixed.

File: modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/InitPhenotypeAnnotation.pm [hunk 4]
```diff
@@ -52,7 +52,12 @@ sub fetch_input {
   unless ($run_type eq NONE) {
     my %import_species = &species;
 
-    if($run_type eq IMPC || $run_type eq MGI || $run_type eq Mouse ){
+    #Mouse, Human and specific AnimalSet have separate init analysis
```
[the remark this answers is not in the document]
Comment (ima23) [reply modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/InitPhenotypeAnnotation.pm]: jira ticket created.

**Comment by ima23** [comment]
vep.t failing, not related to PR

**Pull request merged**